paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -181,28 +181,15 @@ internal class IntentConfirmationHandler(
             return
         }
 
-        val confirmationOption = arguments.confirmationOption
-
-        val requiresPreconfirm = confirmationOption is PaymentConfirmationOption.GooglePay ||
-            confirmationOption.isBacs()
-
-        _state.value = if (requiresPreconfirm) {
-            State.Preconfirming(
-                confirmationOption = arguments.confirmationOption,
-                inPreconfirmFlow = false,
-            )
-        } else {
-            State.Confirming
-        }
+        _state.value = State.Preconfirming(
+            confirmationOption = arguments.confirmationOption,
+            inPreconfirmFlow = false,
+        )
```

Comment:
(non-blocking) should preconfirm also be responsible for setting this state? That's what confirm does

Reply:
I think ideally but we need to ensure that the state is updated in the thread that called `start` to ensure `awaitIntentResult` does not immediately return `null`. I think we can solve this with the executors.